app/components/ProfileComments.jsx

```diff
@@ -0,0 +1,121 @@
+var React = require('react')
+
+var TimeService = require('../modules/time/TimeService.js');
+
+var ReactMarkdown = require('react-markdown');
+
+
+var ProfileComments = React.createClass({
+
+  removeComment: function (commentId) {
+        if (typeof window != 'undefined') {
+            var form = document.createElement("form");
+            form.setAttribute("method", "post");
+            form.setAttribute("action", "/profileCommentRemove");
+            
+            var volonteerIdField = document.createElement("input");
+            volonteerIdField.setAttribute("type", "hidden");
+            volonteerIdField.setAttribute("name", "volonteerId");
+            volonteerIdField.setAttribute("value", this.props.volonteerId);
+            form.appendChild(volonteerIdField);
+            var commentIdField = document.createElement("input");
+            commentIdField.setAttribute("type", "hidden");
+            commentIdField.setAttribute("name", "commentId");
+            commentIdField.setAttribute("value", commentId);
+            form.appendChild(commentIdField);
+            document.body.appendChild(form);
+            form.submit();
+            document.body.removeChild(form);
+        }
+  },
+  getInitialState: function () {
+    return {
+        editId: -1
+    };
+  },
+  editComment: function(commentId) {
+        this.setState({editId: commentId});
+  },
+  cancelEditComment: function () {
+        this.setState({editId: -1});
+  },
+  
+  render: function() {
+    var self = this;
+    var comments = this.props.data.map (function (comment) {
+        if (comment.id == self.state.editId) {
+            return (
+                <tr>
+                    <td colSpan="3">
+                        <form action="/profileCommentUpdate" method="POST">
```

Comment:
To tak niestety nie może być, gdyż cały nasz przepływ zdarzeń i danych odbywa się zgodnie z założeniami architektury [Flux](https://facebook.github.io/flux/docs/overview.html). W skrócie oznacza to że interface (komponent React) może tylko wykonywać (poprzez `this.context.executeAction`) wcześniej zdefiniowane akcje. Zmian w interface może natomiast dokonywać tylko *Store* (zasób danych) który nasłuchuje zdarzeń wywoływanych przez akcje:

![](https://facebook.github.io/flux/img/flux-simple-f8-diagram-with-client-action-1300w.png)

Na prostym przykładzie listy [ToDo](https://github.com/yahoo/flux-examples/blob/master/todo/) usuwanie zadań (lub czegokolwiek) wygląda tak:

Komponent React zawierający obsługę przycisku *destroy*:
```
  destroy: function (todo) {
        this.context.executeAction(deleteTodo, {
            ids: [todo.id]
        });
    }, 
```
źródło: https://github.com/yahoo/flux-examples/blob/master/todo/components/TodoApp.jsx#L72-L76

Definicja akcji `deleteTodo`
```
function (context, payload, done) {
    context.dispatch('DELETE_TODO_START', payload);
    context.service.delete('todo', payload, {}, function (err, todos) {
        // ...
        context.dispatch('DELETE_TODO_SUCCESS', todos);
        done();
    });
};
```
źródło: https://github.com/yahoo/flux-examples/blob/master/todo/actions/deleteTodo.js#L8-L21

I teraz już wystarczy w pliku zasobu danych (Store) dodać obsługę zdarzenia `DELETE_TODO_SUCCESS` wywołując w nim ponowne rysowanie komponentu listy zadań (a w naszym wypadku komentarzy)

przykład: https://github.com/yahoo/flux-examples/blob/master/todo/stores/TodoStore.js#L24-L27

Dzięki temu dostaniemy też obsługę usuwania (i przy okazji edycji) danych po stronie serwera za darmo (dla tego że korzystamy z fluxible) i zmiany w pliku `server.js` staną się zbędne.

W razie wątpliwości pytaj ;)